improver_tests/utilities/copy_attributes/test_CopyAttributes.py

```diff
@@ -15,45 +16,92 @@ class HaltExecution(Exception):
 
 @patch("improver.utilities.copy_attributes.as_cubelist")
 def test_as_cubelist_called(mock_as_cubelist):
+    """Test that the as_cubelist function is called."""
     mock_as_cubelist.side_effect = HaltExecution
     try:
         CopyAttributes(["attribA", "attribB"])(
-            sentinel.cube0, sentinel.cube1, sentinel.template_cube
+            sentinel.cube0, sentinel.template_cube1, sentinel.template_cube2
         )
     except HaltExecution:
         pass
     mock_as_cubelist.assert_called_once_with(
-        sentinel.cube0, sentinel.cube1, sentinel.template_cube
+        sentinel.cube0, sentinel.template_cube1, sentinel.template_cube2
     )
 
 
-def test_copy_attributes_multi_input():
+@pytest.mark.parametrize("history", [False, True])
+def test_copy_attributes_multi_input(history):
     """
-    Test the copy_attributes function for multiple input cubes.
+    Test the copy_attributes function for multiple input template cubes.
 
     Demonstrates copying attributes from the template cube to the input
     cubes and also demonstrates the attributes on the templates cube that
     aren't specified in the attributes list are indeed ignored.
 
+    Also demonstrates that the most recent history attribute is copied correctly if
+    present on multiple template cubes.
+
     Note how we are verifying the object IDs, since CubeAttributes is an
     in-place operation.
     """
     attributes = ["attribA", "attribB"]
     cube0 = Cube([0], attributes={"attribA": "valueA", "attribB": "valueB"})
-    cube1 = Cube([0], attributes={"attribA": "valueAA", "attribB": "valueBB"})
     template_cube = Cube(
-        [0], attributes={"attribA": "tempA", "attribB": "tempB", "attribC": "tempC"}
+        [0],
+        attributes={
+            "attribA": "tempA",
+            "attribB": "tempB",
+            "attribC": "tempC",
+            "history": "2024-11-25T00:00:00Z",
+        },
+    )
+    template_cube_2 = Cube(
+        [0],
+        attributes={
+            "attribA": "tempA",
+            "attribC": "tempC",
+            "history": "2024-11-25T01:43:15Z",
+        },
+    )
+    if history:
+        attributes.append("history")
+
+    plugin = CopyAttributes(attributes)
+    result = plugin.process(cube0, template_cube_2, template_cube)
+    assert type(result) is Cube
```

Comment:
I don't think it really matters but it's probably better to use `assert isinstance(result, Cube)` here

Reply:
Done